### ISSUE TITLE
SASL_SSL is not detected when using multiple listeners with LISTENER_SECURITY_PROTOCOL_MAP

### ISSUE THREAD

**Issue by ssolomon-app** [bitnami/bitnami-docker-kafka#53]
**Description**

Getting the following error when using SASL_SSL enabled named listeners - where the security protocol for each named listener is configured by a KAFKA_CFG_LISTENER_SECURITY_PROTOCOL_MAP environment variable and where the inter broker listener name is defined by a KAFKA_CFG_INTER_BROKER_LISTENER_NAME variable.

`ERROR ==> The KAFKA_CFG_LISTENERS environment variable does not set a secure listener. Set the environment variable ALLOW_PLAINTEXT_LISTENER=yes to allow the container to be started with a plaintext listener. This is recommended only for development.`

The rootfs/libkafka.sh script assumes that SASL gets only enabled through KAFKA_CFG_LISTENERS env variable.

**Steps to reproduce the issue:**

1. Configure internal and external named listeners and use KAFKA_CFG_LISTENER_SECURITY_PROTOCOL_MAP to configure each listener's protocol as shown in the following docker-compose configuration.

`version: '2'

services:
  zookeeper-0:
    image: 'bitnami/zookeeper:3.5.5'
    ports:
      - '2181:2181'
    volumes:
      - "${ZOOKEEPER_VOLUME_DIR}:/bitnami"
    environment:
      - ZOO_SERVER_ID=0
      - ZOO_ENABLE_AUTH=yes
      - ZOO_SERVER_USERS=kafkazookeeper
      - ZOO_SERVER_PASSWORDS=${KAFKA_ZOO_KEEPER_PASSWORD}
      - ZOO_CLIENT_USER=kafkazookeeper
      - ZOO_CLIENT_PASSWORD=${KAFKA_ZOO_KEEPER_PASSWORD}
      - ZOO_HEAP_SIZE=512
  kafka-0:
    image: 'bitnami/kafka:2.2.0'
    ports:
      - '9092:9092' #This port is used to access kafka from the local machine hosting docker
      - '29092:29092' #This port is used to access kafka from within the docker network
      #For more details see: https://rmoff.net/2018/08/02/kafka-listeners-explained/
    volumes:
      - "${KAFKA_VOLUME_DIR}:/bitnami/kafka/"
      - './ssl/kafka.keystore.jks:/opt/bitnami/kafka/conf/certs/kafka.keystore.jks:ro'
      - './ssl/kafka.truststore.jks:/opt/bitnami/kafka/conf/certs/kafka.truststore.jks:ro'
      # Uncomment the following to use customized log levels from ./conf/log4j.properties
      #- './conf/log4j.properties:/opt/bitnami/kafka/conf-tmp/log4j.properties:ro'
    environment:
      - KAFKA_CFG_BROKER_ID=0
      - KAFKA_CFG_BROKER_ID_GENERATION_ENABLE=false
      - KAFKA_CFG_ZOOKEEPER_CONNECT=zookeeper-0:2181
      - KAFKA_CFG_LISTENERS=LISTENER_INTERNAL://kafka-0:29092,LISTENER_EXTERNAL://kafka-0:9092
      - KAFKA_CFG_ADVERTISED_LISTENERS=LISTENER_INTERNAL://kafka-0:29092,LISTENER_EXTERNAL://localhost:9092
      - KAFKA_CFG_LISTENER_SECURITY_PROTOCOL_MAP=LISTENER_INTERNAL:SASL_SSL,LISTENER_EXTERNAL:SASL_SSL
      - KAFKA_CFG_INTER_BROKER_LISTENER_NAME=LISTENER_INTERNAL
      - KAFKA_CFG_AUTO_CREATE_TOPICS_ENABLE=false
      - KAFKA_ZOOKEEPER_USER=kafkazookeeper
      - KAFKA_ZOOKEEPER_PASSWORD=${KAFKA_ZOO_KEEPER_PASSWORD}
      - KAFKA_INTER_BROKER_USER=kafkabroker
      - KAFKA_INTER_BROKER_PASSWORD=${KAFKA_INTER_BROKER_PASSWORD}
      - KAFKA_BROKER_USER=pagakafka
      - KAFKA_BROKER_PASSWORD=${KAFKA_BROKER_PASSWORD}
      - KAFKA_CERTIFICATE_PASSWORD=${KAFKA_CERTIFICATE_PASSWORD}
      - KAFKA_CFG_SSL_ENDPOINT_IDENTIFICATION_ALGORITHM= #disabled`

**Describe the results you received:**

Kafka terminates with:
ERROR ==> The KAFKA_CFG_LISTENERS environment variable does not set a secure listener. Set the environment variable ALLOW_PLAINTEXT_LISTENER=yes to allow the container to be started with a plaintext listener. This is recommended only for development.

**Describe the results you expected:**

SASL_SSL Protocol should be detected and server.properties should be configured accordingly

**Additional information you deem important (e.g. issue happens only occasionally):**

I was able to update 2/debian-9/rootfs/libkafka.sh and get it to work by making the changes shown in the attached patch file
[bitnami-kafka-sasl-interbroker-listener-name-support.patch.txt](https://github.com/bitnami/bitnami-docker-kafka/files/3305906/bitnami-kafka-sasl-interbroker-listener-name-support.patch.txt)

.

**Version**

- Output of `docker version`:

```
Client:
 Version:           18.09.5
 API version:       1.39
 Go version:        go1.10.4
 Git commit:        e8ff056
 Built:             Thu May  9 23:18:36 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          18.09.5
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.4
  Git commit:       e8ff056
  Built:            Thu May  9 22:53:57 2019
  OS/Arch:          linux/amd64
  Experimental:     false

```

- Output of `docker info`:

```
Containers: 3
 Running: 1
 Paused: 0
 Stopped: 2
Images: 851
Server Version: 18.09.5
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 
runc version: N/A
init version: v0.18.0 (expected: fec3683b971d9c3ef73f284f176672c44b448662)
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.15.0-51-generic
Operating System: Ubuntu 16.04.6 LTS
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 15.55GiB
Name: XXXXX
ID: XXXXXX
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Labels:
Experimental: false
Insecure Registries:
 XXXXXX
Live Restore Enabled: false

WARNING: No swap limit support

```

- Output of `docker-compose version` (if applicable):

```
docker-compose version 1.23.2, build 1110ad01
docker-py version: 3.6.0
CPython version: 3.6.7
OpenSSL version: OpenSSL 1.1.0f  25 May 2017
```

**Comment by marcosbc** [comment]
Thanks for the details and the patch @ssolomon-app! Do you wish to send a PR for yourself to make the contribution public? If not we can do it without issues by ourselves.

The only minor comment is, we usually add a space separating comments from text (`#Remove` -> `# Remove`), but apart from that it looks great.

**Comment by ssolomon-app** [comment]
@marcosbc OK I will create a PR. 
Thanks,

**Comment by marcosbc** [comment]
Hi, just wanted to let you know that Kafka was successfully released with your changes! The released container image tags are `2.2.1-debian-9-r30`, `2.2.1-debian-9-r30 `, `1.1.1-debian-9-r240` and `1.1.1-ol-7-r322`.
